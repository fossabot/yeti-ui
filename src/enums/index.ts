export enum ButtonSizes {
  TINY = '0.6rem',
  SMALL = '0.75rem',
  DEFAULT = '0.9rem',
  LARGE = '1.25rem',
}

export enum Breakpoints {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
  XLARGE = 'xlarge',
  XXLARGE = 'xxlarge',
}

export enum BadgeColors {
  INFO = 'info',
  SECONDARY = 'secondary',
  SUCCESS = 'success',
  WARNING = 'warning',
  ALERT = 'alert',
}

export enum ButtonColors {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  SUCCESS = 'success',
  ALERT = 'alert',
  WARNING = 'warning',
}

export enum ButtonGroupColors {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  SUCCESS = 'success',
  ALERT = 'alert',
  WARNING = 'warning',
}

export enum CalloutColors {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  SUCCESS = 'success',
  WARNING = 'warning',
  ALERT = 'alert',
}

export enum LabelColors {
  INFO = 'info',
  SECONDARY = 'secondary',
  SUCCESS = 'success',
  WARNING = 'warning',
  ALERT = 'alert',
}

export enum ProgressColors {
  SECONDARY = 'secondary',
  SUCCESS = 'success',
  WARNING = 'warning',
  ALERT = 'alert',
}

export enum CalloutSizes {
  SMALL = 'small',
  LARGE = 'large',
}

export enum ButtonGroupSizes {
  TINY = 'tiny',
  SMALL = 'small',
  LARGE = 'large',
}

export enum SwitchSizes {
  TINY = 'tiny',
  SMALL = 'small',
  LARGE = 'large',
}

export enum HorizontalAlignments {
  CENTER = 'center',
  RIGHT = 'right',
  JUSTIFY = 'justify',
  SPACED = 'spaced',
}

export enum VerticalAlignments {
  TOP = 'top',
  MIDDLE = 'middle',
  BOTTOM = 'bottom',
  STRETCH = 'stretch',
}

export enum MenuAlignments {
  RIGHT = 'right',
  CENTER = 'center',
}

export enum FloatTypes {
  LEFT = 'left',
  CENTER = 'center',
  RIGHT = 'right',
}

export enum SwitchInputTypes {
  CHECKBOX = 'checkbox',
  RADIO = 'radio',
}

export enum GutterTypes {
  MARGIN = 'margin',
  PADDING = 'padding',
}

export enum ExtendedBreakpoints {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
  XLARGE = 'xlarge',
  XXLARGE = 'xxlarge',
  ALL = 'all',
}

export enum SpaceControls {
  AUTO = 'auto',
  GROW = 'grow',
  SHRINK = 'shrink',
}
