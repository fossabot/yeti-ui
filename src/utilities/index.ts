import {
  merge,
} from 'glamor';
import {
  Breakpoints,
  ExtendedBreakpoints,
  FloatTypes,
  HorizontalAlignments,
  SpaceControls,
  VerticalAlignments,
} from '../enums';

export const listOfButtonProps = [
  'isDropdown',
  'buttonSize',
  'isExpanded',
  'isHollow',
  'isDisabled',
  'isAriaDisabled',
  'isClear',
  'backgroundColor',
  'buttonType',
];

export interface IGeneralPropTypes {
  showFor?: Breakpoints.MEDIUM | Breakpoints.LARGE;
  showOnlyFor?: Breakpoints;
  hideFor?: Breakpoints;
  hideOnlyFor?: Breakpoints;
  isHidden?: boolean;
  isInvisible?: boolean;
  showForLandscape?: boolean;
  showForPortrait?: boolean;
  showForSr?: boolean;
  showOnFocus?: boolean;
  isClearfix?: boolean;
  float?: FloatTypes;
}

export interface IButtonProp extends IGeneralPropTypes {
  buttonType?: string;
  fontFamily?: string;
  padding?: string;
  margin?: string;
  buttonSize?: string;
  fill?: string;
  children?: any;
  color?: string;
  colorAlt?: string;
  radius?: string;
  hollowBorderWidth?: string;
  sizes?: string;
  palette?: string;
  opacityDisabled?: string;
  backgroundHoverLightness?: string;
  hollowHoverLightness?: string;
  transition?: string;
  offset?: string;
  backgroundColor?: string;
  backgroundHover?: string;
  size?: string;
  onClick?: any;
  isHollow?: boolean;
  isClear?: boolean;
  isExpanded?: boolean;
  isDisabled?: boolean;
  isAriaDisabled?: boolean;
  isDropdown?: boolean;
}

export interface ILinkProp extends IButtonProp {
  href: string;
}

export interface IFlexboxPropTypes {
  alignX: HorizontalAlignments;
  alignY: VerticalAlignments;
  selfAlignX: HorizontalAlignments;
  selfAlignY: VerticalAlignments;
  centerAlign: boolean;
  flexContainer: boolean;
  flexDirRow: ExtendedBreakpoints;
  flexDirRowRev: ExtendedBreakpoints;
  flexDirCol: ExtendedBreakpoints;
  flexDirColRev: ExtendedBreakpoints;
  flexChild: SpaceControls;
  flexOrder: number;
  flexOrderSmall: number;
  flexOrderMedium: number;
  flexOrderLarge: number;
}

export enum directions {
  DOWN = 'DOWN',
  UP = 'UP',
  RIGHT = 'RIGHT',
  LEFT = 'LEFT',
}

export function cssTriangle(size: string, color: string, direction: directions) {
  const triangleBase = {
    border: `inset ${size}`,
    content: '\'\'',
    display: 'block',
    height: 0,
    width: 0,
  };
  let triangleDirection;

  if (direction === directions.DOWN) {
    triangleDirection = {
      borderBottomWidth: 0,
      borderColor: `${color} transparent transparent`,
      borderTopStyle: 'solid',
    };
  }
  if (direction === directions.UP) {
    triangleDirection = {
      borderBottomStyle: 'solid',
      borderColor: `transparent transparent ${color}`,
      borderTopWidth: 0,
    };
  }
  if (direction === directions.RIGHT) {
    triangleDirection = {
      borderColor: `transparent transparent transparent ${color}`,
      borderLeftStyle: 'solid',
      borderRightWidth: 0,
    };
  }
  if (direction === directions.LEFT) {
    triangleDirection = {
      borderColor: `transparent ${color} transparent transparent`,
      borderLeftWidth: 0,
      borderRightStyle: 'solid',
    };
  }

  return merge(
    triangleBase,
    triangleDirection,
  );
}

interface IDirectionalSidesObject {
  TOP: string;
  BOTTOM: string;
  LEFT: string;
  RIGHT: string;

  [key: string]: string;
}

export enum sideDirections {
  TOP = 'TOP',
  BOTTOM = 'BOTTOM',
  LEFT = 'LEFT',
  RIGHT = 'RIGHT',
}

export function getSides(val: string, side: sideDirections) {
  const numberOfSides = val.split(' ');

  if (numberOfSides.length === 1) {
    return val;
  }

  if (numberOfSides.length === 2) {
    const twoSidedObject: IDirectionalSidesObject = {
      BOTTOM: numberOfSides[0],
      LEFT: numberOfSides[1],
      RIGHT: numberOfSides[1],
      TOP: numberOfSides[0],
    };

    return twoSidedObject[side];
  }

  if (numberOfSides.length === 3) {
    const twoSidedObject: IDirectionalSidesObject = {
      BOTTOM: numberOfSides[1],
      LEFT: numberOfSides[1],
      RIGHT: numberOfSides[2],
      TOP: numberOfSides[0],
    };

    return twoSidedObject[side];
  }

  if (numberOfSides.length === 4) {
    const twoSidedObject: IDirectionalSidesObject = {
      BOTTOM: numberOfSides[1],
      LEFT: numberOfSides[2],
      RIGHT: numberOfSides[3],
      TOP: numberOfSides[0],
    };

    return twoSidedObject[side];
  }
}

export function removeProps(object: any, remove: string[]) {
  const result: any = {};

  for (const property in object) {
    if (object.hasOwnProperty(property) && remove.indexOf(property) === -1) {
      result[property] = object[property];
    }
  }

  return result;
}
