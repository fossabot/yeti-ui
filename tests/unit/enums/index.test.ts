import * as enums from '../../../src/enums';

describe('src/enums/index.ts', () => {

  test('enums should match snapshot', () => {
    expect(enums).toMatchSnapshot();
  });
});
