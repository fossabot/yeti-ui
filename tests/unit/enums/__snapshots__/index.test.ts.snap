// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/enums/index.ts enums should match snapshot 1`] = `
Object {
  "BadgeColors": Object {
    "ALERT": "alert",
    "INFO": "info",
    "SECONDARY": "secondary",
    "SUCCESS": "success",
    "WARNING": "warning",
  },
  "Breakpoints": Object {
    "LARGE": "large",
    "MEDIUM": "medium",
    "SMALL": "small",
    "XLARGE": "xlarge",
    "XXLARGE": "xxlarge",
  },
  "ButtonColors": Object {
    "ALERT": "alert",
    "PRIMARY": "primary",
    "SECONDARY": "secondary",
    "SUCCESS": "success",
    "WARNING": "warning",
  },
  "ButtonGroupColors": Object {
    "ALERT": "alert",
    "PRIMARY": "primary",
    "SECONDARY": "secondary",
    "SUCCESS": "success",
    "WARNING": "warning",
  },
  "ButtonGroupSizes": Object {
    "LARGE": "large",
    "SMALL": "small",
    "TINY": "tiny",
  },
  "ButtonSizes": Object {
    "DEFAULT": "0.9rem",
    "LARGE": "1.25rem",
    "SMALL": "0.75rem",
    "TINY": "0.6rem",
  },
  "CalloutColors": Object {
    "ALERT": "alert",
    "PRIMARY": "primary",
    "SECONDARY": "secondary",
    "SUCCESS": "success",
    "WARNING": "warning",
  },
  "CalloutSizes": Object {
    "LARGE": "large",
    "SMALL": "small",
  },
  "ExtendedBreakpoints": Object {
    "ALL": "all",
    "LARGE": "large",
    "MEDIUM": "medium",
    "SMALL": "small",
    "XLARGE": "xlarge",
    "XXLARGE": "xxlarge",
  },
  "FloatTypes": Object {
    "CENTER": "center",
    "LEFT": "left",
    "RIGHT": "right",
  },
  "GutterTypes": Object {
    "MARGIN": "margin",
    "PADDING": "padding",
  },
  "HorizontalAlignments": Object {
    "CENTER": "center",
    "JUSTIFY": "justify",
    "RIGHT": "right",
    "SPACED": "spaced",
  },
  "LabelColors": Object {
    "ALERT": "alert",
    "INFO": "info",
    "SECONDARY": "secondary",
    "SUCCESS": "success",
    "WARNING": "warning",
  },
  "MenuAlignments": Object {
    "CENTER": "center",
    "RIGHT": "right",
  },
  "ProgressColors": Object {
    "ALERT": "alert",
    "SECONDARY": "secondary",
    "SUCCESS": "success",
    "WARNING": "warning",
  },
  "SpaceControls": Object {
    "AUTO": "auto",
    "GROW": "grow",
    "SHRINK": "shrink",
  },
  "SwitchInputTypes": Object {
    "CHECKBOX": "checkbox",
    "RADIO": "radio",
  },
  "SwitchSizes": Object {
    "LARGE": "large",
    "SMALL": "small",
    "TINY": "tiny",
  },
  "VerticalAlignments": Object {
    "BOTTOM": "bottom",
    "MIDDLE": "middle",
    "STRETCH": "stretch",
    "TOP": "top",
  },
}
`;
